packages/wallets/src/components/Base/WalletClipboard/__tests__/WalletClipboard.spec.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { useCopyToClipboard, useHover } from 'usehooks-ts';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import useDevice from '../../../../hooks/useDevice';
+import WalletClipboard from '../WalletClipboard';
+
+jest.mock('usehooks-ts', () => ({
+    useCopyToClipboard: jest.fn(),
+    useHover: jest.fn(),
+}));
+
+jest.mock('../../../../hooks/useDevice', () => ({
+    __esModule: true,
+    default: jest.fn(),
+}));
+
+describe('WalletClipboard', () => {
+    let mockCopy: jest.Mock;
+    const mockUseDevice = useDevice as jest.Mock;
+
+    beforeEach(() => {
+        mockCopy = jest.fn();
+        (useCopyToClipboard as jest.Mock).mockReturnValue([null, mockCopy]);
+        (useHover as jest.Mock).mockReturnValue(false);
+        mockUseDevice.mockReturnValue({ isMobile: false });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render the button with copy icon initially', () => {
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
+
+        expect(button).toBeInTheDocument();
+        expect(screen.getByTestId('dt_legacy_copy_icon')).toBeInTheDocument();
+    });
+    it('should render the button with won icon when copied', async () => {
```

Comment:
Nitpick:
According to the best practices it makes sense to not add 'should' word in 'it' blocks b/c spec itself already meant that smth should behave somehow.
For ex. In this case it's probably cleaner to use `renders the button with won icon when copied`